### DISCUSSION TITLE
What should happen if a collaborative close gets rejected?

### DISCUSSION THREAD

**Discussion by da-kami** [get10101/itchysats#1132] [Q&A]
Since merging the event sourcing model we started force-closing positions automatically if a settlement proposal gets rejected by the maker. 

While this seems to be nice in theory (the user wants to close so we ensure it closes...) this might not be in the user's interest. Given that we roll over every hour the user will have a between 24h and 23h waiting period to see the force close going in. So, we should probably not automatically trigger that, but still give the user a choice. 

This would mean, that we have to record a rejected event and then we are "back to open" if a collaborative close gets rejected.

**Comment by thomaseizinger** [comment]
This question gets very interesting if we take the viewpoint that a collaborative close is actually placing an order that nullifies your current position.

In that world, a "rejected collaborative close" is an order that was not filled by the maker. The question here is, given that a maker publishes a price feed (the current offer), what are the reasons for them to not fill the order?

- The taker might have used a different price
- The taker might have provided a quantity that is out of the bounds
- (non-exhaustive list)

Zooming out a bit more, we might want to move away from the notion of "rejecting" an order. On an exchange, placing an order is always successful. It not getting filled is not an error case. Perhaps we should be doing the same thing. If a taker places an order with a maker that doesn't correspond to the parameters, it just remains open. A taker should probably be able to cancel such an order at any time.
